spec/requests/api/v2/providers/courses_spec.rb

```diff
@@ -351,4 +365,54 @@ def perform_request
       end
     end
   end
+
+  describe 'DELETE destroy' do
+    let(:path) do
+      "/api/v2/providers/#{provider.provider_code}" +
+        "/courses/#{course.course_code}"
+    end
+
+    let(:course) do
+      course = create(:course, provider: provider)
+      create(:site_status, :new, course: course)
+      course
+    end
+
```

Comment:
If you change this to 

`
let(:course) { create(:course, provider: provider, site_statuses: [site_status) }
let(:site_status) { build(:site_status, :new) }
    
before do 
  course
end 
`
It won't create the extra course. It only creates the additional course if it is not being built into another object using build.

Reply:
I wasn't aware, thanks a lot for that.

We might be talking past each other here though, in regards to the issue in the [Trello card](https://trello.com/c/ZHWQr2Wq/1839-spec-factories-remove-automatic-creation-of-course-from-sitestatus). I don't think this is creating an extra course. It's older specs unrelated to this PR that are accidentally creating extra courses.